jenkinsapi/jenkins.py

```diff
@@ -346,14 +353,30 @@ def install_plugins(self, plugin_list, restart=False,
         if restart and self.plugins.restart_required:
             self.safe_restart(wait_for_reboot=wait_for_reboot)
 
-    def delete_plugin(self, plugin):
+    def delete_plugin(self, plugin, restart=False, wait_for_reboot=False):
+        """
+        Delete a plugin and optionally restart jenkins. Will not delete
+        dependencies.
+        @param plugin: Plugin to be deleted
+        @param restart: Boolean, restart jenkins after plugin deletion
+        """
         if isinstance(plugin, Plugin):
             plugin = plugin.shortName
         del self.plugins[plugin]
+        if restart and self.plugins.restart_required:
+            self.safe_restart(wait_for_reboot=wait_for_reboot)
 
-    def delete_plugins(self, plugin_list):
+    def delete_plugins(self, plugin_list, restart=False, wait_for_reboot=False):
+        """
+        Delete a list of plugins and optionally restart jenkins. Will not delete
+        dependencies.
+        @param plugin_list: List of plugins to be deleted
```

Comment:
Please say here that it is a List of `Plugin` or list of names of plugins. It would be a bit clearer.

Reply:
That's the thing: it doesn't really matter. The method `delete_plugin` will automatically convert it to strings. I can still change it, should you want.